test/e2e/kuma_test.go

```diff
@@ -144,6 +144,10 @@ func TestDeployAllInOnePostgresKuma(t *testing.T) {
 	deployIngress(ctx, t, env)
 	service, err := env.Cluster().Client().CoreV1().Services("default").Get(ctx, "httpbin", metav1.GetOptions{})
 	require.NoError(t, err)
+
+	if service.ObjectMeta.Annotations == nil {
```

Comment:
While this might fix this, I'm not sure I understand the situation in which this will happen. The service has an empty annotations map (hence `nil`) or it doesn't exist so the `.Get()` returns no error and an "empty" service?

Reply:
if the service does not exist, `Get()` would return a `not found` error. So the service would never be `nil` when no error returned from `Get`.

Reply:
So what's the situation when `.Annotations` are `nil`? Is that expected? Or that happens "randomly"?

What I'm trying to say is that if we're trying to work around a flaw in the system let's document it in a comment. If that's unexpected then... well exactly: what then?

Is there something that sets those annotations and does not manage to do it in time by the time we're hitting the `.Get`?

Reply:
Unresolving this as I believe there's value in understanding **why** the annotations are unset.

Reply:
Ok, so it seems it was me not knowing the full context of this. Thanks for looking into this @shaneutt 🙇